File: .cline/skills/playwright-fieldkit/templates/test-data-lifecycle.py
"""Adapt this fixture to the application's API and data factories."""

from collections.abc import Iterator
from uuid import uuid4

import pytest


@pytest.fixture
def test_entity(request) -> Iterator[dict[str, str]]:
    run_id = uuid4().hex
    entity = {"id": run_id, "email": f"qe+{run_id}@example.test"}

    # Arrange through the project's API/client rather than unrelated UI steps.
    # api.create_entity(entity)
    yield entity

    # Cleanup must be safe to call even after a partially failed test.
    # api.delete_entity_if_exists(entity["id"])

File: .cline/skills/playwright-fieldkit/templates/test-data-lifecycle.ts
import { test as base } from '@playwright/test';
import { randomUUID } from 'node:crypto';

type TestData = { entity: { id: string; email: string } };

export const test = base.extend<TestData>({
  entity: async ({ request }, use) => {
    const id = randomUUID();
    const entity = { id, email: `qe+${id}@example.test` };

    // Arrange through the application's API rather than unrelated UI steps.
    // await request.post('/api/entities', { data: entity });
    await use(entity);

    // Cleanup must be idempotent because the test may fail midway.
    // await request.delete(`/api/entities/${id}`);
  },
});
